### PR TITLE
Add additional information to Safari quirks

### PR DESCRIPTION
**Pull request by P4sca1**
> Testing my Passkey / WebAuthn implementation on Safari 17.2.1 revealed the following:
> 
> * Using a wrapper around `fetch()` **does not work**. In my specific case this was [ofetch](https://github.com/unjs/ofetch), which provides a unified fetch API across Node.js and the browser (but wraps the native browser fetch). You have to call the native fetch directly.
> * It is allowed to await other promises / make fetch requests (even using the wrapper), **after** the call to `startAuthentication`. Before, you should only make a single asynchronous call via the native fetch.
> * Conditional UI is also limited by the User gesture requirement. This is normally not a problem due to the "freebie" call per page you get since Safari 16. However, when using an SPA (e.g. vue-router, react-router, ...), it is not possible to initiate a Conditional UI twice, without hard reloading the page. In my case, this happens when the user opens the login page, switches to another page, and then opens the login page a 2nd time.
> * Using session replay scripts (e.g. Sentry Session Replay, LogRocket, ...) also interferes with user gesture detection. Maybe because they also wrap the native fetch?

https://github.com/MasterKale/SimpleWebAuthn/discussions/433#discussioncomment-8117648

### CODE REVIEW

File: docs/advanced/browser-quirks.md [hunk 1]
```diff
@@ -6,8 +6,10 @@ title: Browser Quirks
...
 
-* [@simplewebauthn/browser's `startRegistration()`](packages/browser.mdx#startregistration) and [`startAuthentication`](packages/browser.mdx#startauthentication) must be called in a **native** click listener. Some JS UI libraries, like VueJS, may not use native click handling without the use of framework-specific functionality [like Vue's `.native` modifier](https://vuejs.org/v2/guide/components-custom-events.html#Binding-Native-Events-to-Components).
-* ~~Safari only supports the use of XHR and `fetch` within click handlers for requesting WebAuthn registration and authentication options.~~ (Fixed as of February 2021. See [here](https://bugs.webkit.org/show_bug.cgi?id=213595))
+* [@simplewebauthn/browser's `startRegistration()`](packages/browser.mdx#startregistration) and [`startAuthentication`](packages/browser.mdx#startauthentication) must be called in a **native** click listener (user gesture detection). Some JS UI libraries, like VueJS 2, may not use native click handling without the use of framework-specific functionality [like Vue's `.native` modifier](https://vuejs.org/v2/guide/components-custom-events.html#Binding-Native-Events-to-Components).
```
Comment (MasterKale) [inline docs/advanced/browser-quirks.md:9]: It turns out Vue 2 is deprecated!

Screenshot 2024-01-13 at 10 43 37 AM

For sake of editorial longevity let's make it a more general reference to the framework:

```suggestion
* [@simplewebauthn/browser's `startRegistration()`](packages/browser.mdx#startregistration) and [`startAuthentication`](packages/browser.mdx#startauthentication) must be called in a **native** click listener (user gesture detection). Some JS UI libraries, like Vue, may not use native click handling without the use of framework-specific functionality [like Vue's `v-on` directive on a native HTML element](https://vuejs.org/api/built-in-directives.html#v-on).
```

I also tossed in an updated reference to how Vue 3 seems to enable listening to native DOM events (though I haven't tested it myself 🤔.) https://vuejs.org/guide/components/events.html#emitting-and-listening-to-events is the latest version of the current URL but it doesn't have anything like `.native` anymore for me to reference so I'm not sure if it's the true updated technique.

Comment (P4sca1) [reply docs/advanced/browser-quirks.md:9]: Vue 3 dropped the .native modifier. Event listeners like @click are now native by default if not manually handled by the component.

Comment (MasterKale) [reply docs/advanced/browser-quirks.md:9]: Unless you are a Vue developer I'm thinking the Vue reference gets taken out completely. I certainly am not keeping up with Vue's evolution that closely to know what to replace the `.native` reference with.

```suggestion
* [@simplewebauthn/browser's `startRegistration()`](packages/browser.mdx#startregistration) and [`startAuthentication`](packages/browser.mdx#startauthentication) must be called in a **native** click listener (user gesture detection). Some JS UI libraries may not use native click handling without the use of framework-specific functionality.
```

Comment (P4sca1) [reply docs/advanced/browser-quirks.md:9]: I agree, that we should remove framework specific information.

**Comment by MasterKale** [comment]
Thanks for contributing to the docs @P4sca1! I remember seeing this behavior a week or two ago, especially the one about a **single** call to `.create()` or `.get()` per page load (which totally wrecks any login page trying to use conditional UI _and_ support modal UI 😱), it's good to have it documented somewhere 🙇‍♂️

**Pull request merged**